src/ts/helper/AdHelper.ts

```diff
@@ -0,0 +1,108 @@
+import { Ad, AdBreak, AdBreakEvent, AdData, AdEvent, ErrorEvent, LinearAd, PlayerAPI, VastAdData } from 'bitmovin-player';
+import * as Conviva from '@convivainc/conviva-js-coresdk';
+
+export class AdHelper {
+  public static mapAdPosition(
+    adBreak: AdBreak,
+    player: PlayerAPI,
+  ): Conviva.valueof<Conviva.ConvivaConstants['AdPosition']> {
+    if (adBreak.scheduleTime <= 0) {
+      return Conviva.Constants.AdPosition.PREROLL;
+    }
+
+    if (adBreak.scheduleTime >= player.getDuration()) {
+      return Conviva.Constants.AdPosition.POSTROLL;
+    }
+
+    return Conviva.Constants.AdPosition.MIDROLL;
+  }
+
+  public static formatAdErrorEvent(event: ErrorEvent & {
+    data?: {
+      code?: number,
+    },
+  }) {
+    const message = event?.message || 'Unknown message';
+    const name = event?.name || 'Unknown name';
+    const formattedErrorParts = [
+      'Ad error:',
+      `${name};`,
+      event.data?.code ? `Ad error code: ${event.data?.code};` : undefined,
+      `Message:`,
+      `${message};`,
+      `Error code: ${event.code};`,
+      event.troubleShootLink ? `Troubleshoot link: ${event.troubleShootLink}` : undefined,
+    ].filter(Boolean);
+
+    return formattedErrorParts.join(' ');
+  }
+
+  public static extractConvivaAdInfo(player: PlayerAPI, adBreakEvent: AdBreakEvent, adEvent: AdEvent): Conviva.ConvivaMetadata {
+    const adPosition = AdHelper.mapAdPosition(adBreakEvent.adBreak, player);
+    const ad = adEvent.ad as Ad | LinearAd;
+    const adData = ad.data as undefined | AdData | VastAdData;
+
+    let adSystemName = 'NA';
+    let creativeId = 'NA';
+    let adTitle: string | undefined;
+    let firstAdId = ad.id;
+
+    // TODO these two are not exposed currently. Add them whenever the player
+    // exposes them similar to https://github.com/bitmovin-engineering/player-android/pull/3147.
+    // Related discussion https://bitmovin.slack.com/archives/C0LJ16JBS/p1716801796326889.
```

Comment:
This is an open source repository. Please remove those internal links.

```suggestion
    // TODO these two properties are not exposed currently. Add them whenever the player
    // exposes them.
```

Reply:
I kept them intentionally although this is an open source project. Those links are protected and the related discussion is important, why do you think they need to be removed? I don't think it's a secret that Bitmovin uses Slack. 🙂

Reply:
No, that's definitely not a secret.

I just thought that if someone outside of Bitmovin wants to check if those values are supported, he will not have a chance to do so using those links and will need to fall back to publicly available documentation (such as API docs).

Reply:
As we discussed on Slack, I've left it as it is to keep the important references.